### DISCUSSION TITLE
SurrealDB

### DISCUSSION THREAD

**Discussion by CanRau** [ontola/atomic-server#513] [Ideas]
Hey just now stumbled upon this project via the reference from the [Tauri Mobile issue](https://github.com/tauri-apps/tauri/issues/843).
Haven't looked too much into it, though it looks really interesting, especially as I have something (maybe) similar in my todos, so far only collecting ideas etc 🥲

I noticed that you're using sled, which is also on my list as an early option, after I think BadgerDB is too memory heavy from what I've read.

Then recently stumbled upon [SurrealDB](https://surrealdb.com/) which might be an interesting alternative, with I think, more and very interesting features for such a project.

Curious if you know it and what you think.

Also, just as a disclaimer, so far I don't have any personal Rust dev experience 😅

Maybe it might even be interesting to talk about your plans (also should look more into your content to learn more) and maybe even collab, don't know just thinking out loud 🤓 Though I'm pretty busy right now with a startup we wanna launch in january so yea 🙈

**Comment by joepio** [comment]
Hi @CanRau!

Thanks for the kind words :)

SurrrealDB is a really cool piece of software, but I think it's not really a suitable replacement for Sled in our case. Sled is designed to be embedded and is far more low-level: it's just a KV store. Sled doesn't have a query language, authorization, a JS engine and all that - it's far smaller. That is great for Atomic-Server, because I want to keep it as simple, lightweight and fast as possible.

I think SurrealDB and Atomic-Server can be thought as alternatives for each other, in a way. They both provide various indexes, allowing users to quickly filter and sort their data. They both have authorization mechanisms. They both provide a RESTful API layer. They both have updates over websockets. In various ways SurrealDB is further than Atomic-Server, such as supporting mult-node setups and embedding JS + WASM functions.

Where Atomic Server is probably a bit more advanced, is regarding its UI and decentralized features. Surreal doesn't have a GUI at this moment (although it is being developed), whereas Atomic-Server has full on collaborative documents, group chats, tables and more. 

Would love to have a chat! [Join our discord](https://discord.gg/a72Rv2P) and let's do just that :)
